src/hooks/scriptLoader.ts

```diff
@@ -0,0 +1,29 @@
+export default function scriptLoader(scriptSrc: string, options?: ScriptLoadOptions): Promise<void> {
+  return new Promise((resolve, reject) => {
+    const script = document.createElement("script")
+    script.type = "text/javascript"
+    script.src = scriptSrc
+    script.async = true
+    script.onload = () => resolve()
+    script.onerror = () => reject()
```

Comment:
Can it be just `script.onload = resolve`?

Reply:
technically yes, but the assignment gives me a a TS error
also this way the Promise payload is empty as indicated by the type